test/js/web/fetch/fetch.test.ts

```diff
@@ -1816,3 +1816,84 @@ it("304 not modified with 0 content-length does not cause a request timeout", as
   expect(await response.arrayBuffer()).toHaveLength(0);
   server.stop(true);
 });
+
+describe("http/1.1 response body length", () => {
+  // issue #6932 (support response without Content-Length and Transfer-Encoding) + some regression tests
+
+  let server: net.Server | undefined;
+  beforeAll(async () => {
+    server = net.createServer(socket => {
+      socket.on("data", data => {
+        const lines = data.toString().split("\r\n");
+        for (const line of lines) {
+          const [method, path] = line.split(" ");
+          if (method === "GET" || method === "HEAD") {
+            if (path === "/text") {
+              socket.end("HTTP/1.1 200 OK\r\nContent-Type: text/plain\r\n\r\nHello, World!");
+            } else if (path === "/json") {
+              socket.end('HTTP/1.1 200 OK\r\nContent-Type: application/json\r\n\r\n{"hello":"World"}');
+            } else if (path === "/chunked") {
+              socket.end(
+                "HTTP/1.1 200 OK\r\nContent-Type: text/plain\r\nTransfer-Encoding: chunked\r\n\r\nd\r\nHello, World!\r\n0\r\n\r\n",
+              );
+            } else if (path === "/empty") {
+              socket.end("HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n");
+            } else {
+              const status = +path.slice(1) || 200;
+              socket.end(
+                `HTTP/1.1 ${status} STATUS MESSAGE\r\nContent-Type: text/plain\r\nContent-Length: 13\r\n\r\nHello, World!`,
+              );
+            }
+          }
+        }
+      });
+    });
+
+    const { promise: start, resolve } = Promise.withResolvers();
+    server.listen(8080, resolve);
```

Comment:
To prevent flaky tests, we should make this change, so it picks a random, open port:
```suggestion
    server.listen(0, resolve);
```